src/browsercheck.js

```diff
@@ -10,20 +10,23 @@ const browsercheck: () => Promise<void> = () => {
 
   return new Promise((resolve, reject) => {
     Modernizr.on('indexeddb', (indexedDBAvailable) => {
-      if (
-        !indexedDBAvailable ||
+      Modernizr.on('workerscrypto', (workerscryptoDBAvailable) => {
+        if (
+          !indexedDBAvailable ||
         !Modernizr.cookies ||
         !Modernizr.getrandomvalues ||
         !Modernizr.filereader ||
         !Modernizr.localstorage ||
-        !Modernizr.webworkers
-      ) {
-        return reject(
-          new Error('Some of required browser APIs are not available')
-        )
-      }
+        !Modernizr.webworkers ||
+        !workerscryptoDBAvailable
+        ) {
+          return reject(
+            new Error('Some of required browser APIs are not available')
+          )
+        }
 
-      return resolve()
+        return resolve()
+      })
```

Comment:
to make these checks clean we need to resolve Promise for each of them, like
```
return checkFaturesSync
  .then(checkIndexedDB)
  .then(checkWorkerRNG)
```

Reply:
Better to have helper function that gets list of async tests and fires once all of them are checked. Like:

```
function checkAll (testNames, callback) {
  Promise.all(
    testNames.map((name) =>
      new Promise((resolve, reject) => 
        Modernizr.on(name, (result) => result ? resolve() : reject())
      )
    )
  )
}
```

And call it like:
```
checkAll(['indexeddb', 'workerscrypto'])
  .then(...)
```

In this case, when you add another async check, you need to add only its name to the list. This also means we could pass list of checks via configuration.

But please don't fix it right now, it's fine this way too.